include/deal.II/matrix_free/shape_info.templates.h

```diff
@@ -95,6 +95,9 @@ namespace internal
       Assert(fe->n_components() == 1,
              ExcMessage("FEEvaluation only works for scalar finite elements."));
 
+      if ((fe->n_dofs_per_cell() == 0) || (quad.size() == 0))
```

Comment:
Look reasonable. Although I would move this down after the `univariate_shape_data` has been initialized, i.e., after `univariate_shape_data.n_q_points_1d = quad.size();`. I think there are multiple places in the library where we do something like `shape_infe.data[0].quadrature`, aren't there? @kronbichler What is your opinion?

Reply:
I agree, we can still set up some data structures with empty content.